src/som/interpreter/actors/EventualMessage.java

```diff
@@ -20,6 +21,7 @@
   protected final Object[]  args;
   protected final SResolver resolver;
   protected final RootCallTarget onReceive;
+  protected final EventualMessage causalMessage;
   protected final long causalMessageId;
```

Comment:
But `EventualMessage` does not have an `id` field, unless we use EventualMessage.getCurrentExecutingMessageId(), but I think that this method will return the `id` of the current executing message (as the name says), if we want the id of the causal message we should save it when is execute it, right? This is used in the TraceBuffer class.